re.c

```diff
@@ -4194,6 +4194,39 @@ rb_reg_s_union_m(VALUE self, VALUE args)
     return rb_reg_s_union(self, args);
 }
 
+/*
+ *  call-seq:
+ *    Regexp.linear_time?(re)
+ *    Regexp.linear_time?(string, options = 0)
+ *
+ *  Returns +true+ if matching against <tt>re</tt> can be
+ *  done in linear time to the input string.
+ *
+ *    Regexp.linear_time?(/re/) # => true
```

Comment:
Please tell me if this documentation is sufficient.

Reply:
I think it's good enough.